.github/workflows/ci.yml

```diff
@@ -97,22 +97,23 @@ jobs:
           manylinux: auto
           args: --manifest-path examples/simple/Cargo.toml
 
-  valgrind:
+  careful-valgrind:
     runs-on: ubuntu-22.04
     needs: [lint, check-msrv, examples]
     steps:
       - uses: actions/checkout@v3
       - uses: actions-rs/toolchain@v1
         with:
-          toolchain: stable
+          toolchain: nightly
           profile: minimal
           default: true
       - uses: Swatinem/rust-cache@v2
         continue-on-error: true
       - run: |
             sudo apt-get install --yes valgrind
```

Comment:
Use `- uses: taiki-e/install-action@valgrind` might be better because it will install the latest version via Snapcraft.